context/src/bazel_bep/common.rs

```diff
@@ -2,44 +2,54 @@ use std::collections::HashMap;
 
 use anyhow::{Ok, Result};
 use bazel_bep::types::build_event_stream::{
-    build_event::Payload, build_event_id::Id, file::File::Uri, BuildEvent,
+    build_event::Payload, build_event_id::Id, file::File::Uri, BuildEvent, TestSummary,
 };
+use chrono::DateTime;
 
-use crate::junit::junit_path::{JunitReportFileWithStatus, JunitReportStatus};
+use crate::junit::junit_path::{
+    JunitReportFileWithTestRunnerReport, TestRunnerReport, TestRunnerReportStatus,
+};
 
 const FILE_URI_PREFIX: &str = "file://";
 
 #[derive(Debug, Clone, Default)]
-pub struct TestResult {
+pub struct BepTestResult {
     pub label: String,
     pub cached: bool,
     pub xml_files: Vec<String>,
-    pub summary_status: Option<JunitReportStatus>,
+    pub test_runner_report: Option<TestRunnerReport>,
 }
 
 #[derive(Debug, Clone, Default)]
 pub struct BepParseResult {
     pub bep_test_events: Vec<BuildEvent>,
     pub errors: Vec<String>,
-    pub test_results: Vec<TestResult>,
+    pub test_results: Vec<BepTestResult>,
 }
 
 impl BepParseResult {
     pub fn from_build_events<I: IntoIterator<Item = Result<BuildEvent>>>(
         events: I,
     ) -> Result<Self> {
-        let (errors, test_results, summary_statuses, bep_test_events) = events.into_iter().fold(
-            (
-                Vec::<String>::new(),
-                Vec::<TestResult>::new(),
-                HashMap::<String, JunitReportStatus>::new(),
-                Vec::<BuildEvent>::new(),
-            ),
-            |(mut errors, mut test_results, mut summary_statuses, mut bep_test_events),
-             parse_event| {
+        #[derive(Debug, Default)]
+        struct Acc {
+            errors: Vec<String>,
+            test_results: Vec<BepTestResult>,
+            test_runner_reports: HashMap<String, TestRunnerReport>,
+            bep_test_events: Vec<BuildEvent>,
+        }
+        let Acc {
+            errors,
+            test_results,
+            test_runner_reports,
+            bep_test_events,
+        } = events
+            .into_iter()
+            .fold(Acc::default(), |mut acc, parse_event| {
```

Comment:
Using a tuple was getting a bit unwieldy here. I switched to a simple local struct since nothing needs access to the this struct and it helps with readability